Added "Documentation Overview" to Working with model documentation

validbeck opened the pull request:
## Internal Notes for Reviewers
> For sc-4038, I added a `Document Overview` section on "Working with model documentation".

### Working with model documentation

#### Looking to get started as a model developer?
Made a small cosmetic change to this call-out to adhere to our style guide emphasis rules.

| Old | New |
|---|---|
|<img width="983" alt="Screenshot 2024-05-21 at 5 05 03 PM" src="https://github.com/validmind/documentation/assets/164545837/462b0147-a137-4f13-aaf4-367a97413c39">|<img width="985" alt="Screenshot 2024-05-21 at 5 05 09 PM" src="https://github.com/validmind/documentation/assets/164545837/9920c1eb-1da9-4aa8-becf-3802fc918ece">|

#### Document Overview
Modified the old `Get started` section into:
- Introduction and content summary of the **Document Overview**
- `Add or edit documentation` (Second half of the old **Get started** section)

| Old | New |
|---|---|
|<img width="796" alt="Screenshot 2024-05-21 at 12 20 02 PM" src="https://github.com/validmind/documentation/assets/164545837/79178924-fe07-4e85-aca2-5e2cbf8276d8">|<img width="890" alt="Screenshot 2024-05-21 at 5 06 10 PM" src="https://github.com/validmind/documentation/assets/164545837/f33a1a4c-fdbd-4648-84b4-2f5e3d2504a4">|

### Assign documentation section statuses
Changed references to "documentation overview" to **Document Overview** as this is the official title in the UI.

validbeck commented:
I've made some changes to the `Document Overview` section as I noticed sc-3817 was related to this current Story. :) 

| Old | New |
|---|---|
|<img width="890" alt="Screenshot 2024-05-21 at 5 06 10 PM" src="https://github.com/validmind/documentation/assets/164545837/f33a1a4c-fdbd-4648-84b4-2f5e3d2504a4">|<img width="665" alt="Screenshot 2024-05-22 at 9 38 47 AM" src="https://github.com/validmind/documentation/assets/164545837/07f42889-d2ff-4dba-b2b9-fd7c459f2e6f">|

nrichers commented on site/guide/documentation-section-statuses.qmd:
```diff
@@ -0,0 +1,34 @@
+    - Done
+
+   ::: {.callout-tip appearance="minimal"}
+    The documentation completion Status summary will then be updated on your Document Overview.
```
```suggestion
    The documentation completion status summary will then be updated on your Document Overview page.
```

AFAIK, "Status" is not a proper noun and "Document Overview" is the name of a page, so it might help to just include that. 

(Not a strong opinion, but I find that page name a bit confusing as it's not shown when you first click on the overview; the bread crumb just refers to "Documentation" the name "Document Overview" is visible only when you are in subsections.)

nrichers reviewed (approved):
LGTM, with some minor comments to address. Let me know if you want to chat about conventions to make sure we're both consistent. 🚢

validbeck commented:
> Thank you for making these changes! You marked this as "changes requested" but what do you want me to update, @validbeck? Your comment doesn't say.

**[Done!](https://github.com/validmind/documentation/blob/beff28c2ad4e7561ec874980ae2c1e99f23d6c0f/site/guide/assign-documentation-section-statuses.qmd)**

> - Avoid "can" if you can, just state the action (advice from a long-time editor I worked with).
> - In this case, "Status" isn't a proper noun, so maybe just a plain reference to "the status drop down" is enough, without bolding? 

- Changed as many of the "can"s to just the action as I could swing
- Done!

> Musing on writing minimalism ... I've also had editors suggest not to use language that refers to senses not everyone has access to 

Agreed after consideration, removed.

> - AFAIK, "Status" is not a proper noun and "Document Overview" is the name of a page, so it might help to just include that.
> - Not a strong opinion, but I find that page name a bit confusing as it's not shown when you first click on the overview

- Done!
- Also agreed 😆 

| Just one can now :p (aka all the cans I have left before I can't) | Simplified the status drop-down instructions |
|---|---|
|<img width="669" alt="Screenshot 2024-05-22 at 2 15 00 PM" src="https://github.com/validmind/documentation/assets/164545837/92733b35-0df3-4dac-821a-439966c44d5c">|<img width="988" alt="Screenshot 2024-05-22 at 2 15 22 PM" src="https://github.com/validmind/documentation/assets/164545837/95cb0781-7cba-4897-9aac-80481ab824ef">|
